internal/node/node.go

```diff
@@ -0,0 +1,75 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type Node interface {
+	IsNodeSchedulable(node *v1.Node) bool
+	GetNodesListBySelector(ctx context.Context, mod *kmmv1beta1.Module) ([]v1.Node, error)
+	GetNumTargetedNodes(ctx context.Context, selectorLabels map[string]string) (int, error)
+}
+
+type node struct {
+	client client.Client
+}
+
+func NewNode(client client.Client) Node {
+	return &node{
+		client: client,
+	}
+}
+
+func (n *node) IsNodeSchedulable(node *v1.Node) bool {
+	for _, taint := range node.Spec.Taints {
+		if taint.Effect == v1.TaintEffectNoSchedule {
+			return false
+		}
+	}
+	return true
+}
+
+func (n *node) GetNodesListBySelector(ctx context.Context, mod *kmmv1beta1.Module) ([]v1.Node, error) {
+	logger := log.FromContext(ctx)
+	logger.V(1).Info("Listing nodes", "selector", mod.Spec.Selector)
+
+	selectedNodes := v1.NodeList{}
+	opt := client.MatchingLabels(mod.Spec.Selector)
+	if err := n.client.List(ctx, &selectedNodes, opt); err != nil {
+		return nil, fmt.Errorf("could not list nodes: %v", err)
+	}
+	nodes := make([]v1.Node, 0, len(selectedNodes.Items))
+
+	for _, node := range selectedNodes.Items {
+		if n.IsNodeSchedulable(&node) {
+			nodes = append(nodes, node)
+		}
+	}
+	return nodes, nil
+}
+
+func (n *node) GetNumTargetedNodes(ctx context.Context, selectorLabels map[string]string) (int, error) {
+	logger := log.FromContext(ctx)
+
+	logger.V(1).Info("Listing nodes", "selector", selectorLabels)
+
+	selectedNodes := v1.NodeList{}
+	opt := client.MatchingLabels(selectorLabels)
+	if err := n.client.List(ctx, &selectedNodes, opt); err != nil {
+		return 0, fmt.Errorf("could not list nodes: %v", err)
+	}
+
+	numNodes := 0
+
+	for _, node := range selectedNodes.Items {
+		if n.IsNodeSchedulable(&node) {
+			numNodes += 1
+		}
+	}
+	return numNodes, nil
```

Comment:
```suggestion
        numTargetedNode, err := n.GetNodesListBySelector(ctx, selector)
        if err != nil {
            return 0, fmt.Errorf("could not list nodes: %v", err)
        }
        return numTargetedNode, nil
```

Reply:
@ybettan `GetNodesListBySelector` returns `[]v1.Node`.
so in the return section of `GetNumTargetedNodes` i will return its length, is that ok?

Reply:
I had a typo, it should be:
```
return len(numTargetedNode), nil
```

I have fixed the suggenstion.

Reply:
@ybettan done